event.go

```diff
@@ -42,12 +46,18 @@ var clock = time.Now
 
 func newRequest(id string, r *http.Request, options *Options) *request {
 	var body any
+
+	if r.Body == nil {
+		r.Body = http.NoBody
+	}
+
```

Comment:
Running into an issue that'll segfault in duplicateBody on a network error, this is after I initialize the Supergood client by default.

Reply:
It'd probably be best not to change the value of `r.Body`, but instead have `duplicateBody` handle this case by returning `nil, nil`.

Reply:
Yeah - good call!